packages/server/src/api/controllers/ai/agents.ts

```diff
@@ -57,11 +167,108 @@ export async function remove(ctx: UserCtx<void, void>) {
 export async function fetchHistory(
   ctx: UserCtx<void, FetchAgentHistoryResponse>
 ) {
-  const db = context.getGlobalDB()
+  const db = context.getAppDB()
   const history = await db.allDocs<AgentChat>(
     docIds.getDocParams(DocumentType.AGENT_CHAT, undefined, {
       include_docs: true,
     })
   )
   ctx.body = history.rows.map(row => row.doc!)
 }
+
+export async function fetchToolSources(
+  ctx: UserCtx<void, AgentToolSourceWithTools[]>
+) {
+  const db = context.getAppDB()
+  const toolSources = await db.allDocs<AgentToolSource>(
+    docIds.getDocParams(DocumentType.AGENT_TOOL_SOURCE, undefined, {
+      include_docs: true,
+    })
+  )
+
+  ctx.body = toolSources.rows.map(row => {
+    const doc = row.doc!
+    let tools: any[] = []
+
+    switch (doc.type) {
```

Comment:
I've seen another switch like this somewhere else in the codebase, and it's a pattern we follow elsewhere too (e.g. FieldTypes) and it leads to bugs. Someone inevitably forgets to update one of the switch statements, which invariably spread throughout the codebase as the features grow. I'd much prefer we centralised these switch statements into some sort of abstract class you have to implement if you're making a ToolSource.